src/components/pages/CompanySubscriptions/index.tsx

```diff
@@ -0,0 +1,320 @@
+import UnpublishedIcon from '@mui/icons-material/Unpublished'
+import {
+  CompanySubscriptionFilterType,
+  type SubscribedActiveApps,
+  SubscriptionStatus,
+} from 'features/apps/types'
+
+interface FetchHookArgsType {
+  statusFilter: string
+  expr: string
+}
+
+export default function CompanySubscriptions() {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  const navigate = useNavigate()
+  const [refresh, setRefresh] = useState(0)
+  const [searchExpr, setSearchExpr] = useState<string>('')
+  const [fetchHookArgs, setFetchHookArgs] = useState<FetchHookArgsType>()
+  const [filterStatus, setFilterStatus] = useState<string>('')
+  const searchInputData = useSelector(updateApplicationRequestSelector)
+  const [group, setGroup] = useState<string>(
+    t('content.companySubscriptions.filter.showAll')
+  )
+  const [showUnsubscribeOverlay, setShowUnsubscribeOverlay] =
+    useState<boolean>(false)
+  const [loading, setLoading] = useState<boolean>(false)
+  const [appId, setAppId] = useState<string>('')
+  const [subscriptionId, setSubscriptionId] = useState<string>('')
+  const [unsubscribeMutation] = useUnsubscribeAppMutation()
+  const [enableErrorMessage, setEnableErrorMessage] = useState<boolean>(false)
+
+  const setView = (e: React.MouseEvent<HTMLInputElement>) => {
+    const viewValue = e.currentTarget.value
+    setFilterStatus(viewValue)
+    setGroup(viewValue)
+    setRefresh(Date.now())
+  }
+
+  const filterView = [
+    {
+      buttonText: t('content.companySubscriptions.filter.requested'),
+      buttonValue: CompanySubscriptionFilterType.REQUESTED,
+      onButtonClick: setView,
+    },
+    {
+      buttonText: t('content.companySubscriptions.filter.active'),
+      buttonValue: CompanySubscriptionFilterType.ACTIVE,
+      onButtonClick: setView,
+    },
+    {
+      buttonText: t('content.companySubscriptions.filter.showAll'),
+      buttonValue: CompanySubscriptionFilterType.SHOW_ALL,
+      onButtonClick: setView,
+    },
+  ]
+
+  useEffect(() => {
+    if (onValidate(searchExpr)) {
+      setFetchHookArgs({
+        statusFilter: filterStatus,
+        expr: searchExpr,
+      })
+    }
+  }, [filterStatus, searchExpr])
+
+  const onValidate = (expr: string) => {
+    const validateExpr = isName(expr)
+    if (validateExpr) dispatch(setSearchInput({ open: true, text: expr }))
+    return validateExpr
+  }
+
+  const onUnsubscribeSubmit = async () => {
+    setLoading(true)
+    await unsubscribeMutation(subscriptionId)
+      .unwrap()
+      .then(() => {
+        success(t('content.organization.unsubscribe.unsubscribeSuccess'))
+        setLoading(false)
+        setShowUnsubscribeOverlay(false)
+        setEnableErrorMessage(false)
+        setRefresh(Date.now())
+      })
+      .catch(() => {
+        setLoading(false)
+        setEnableErrorMessage(true)
+      })
+  }
+
+  const renderStatusButton = (status: string) => {
```

Comment:
could have created common component to render this in both the files

Reply:
These function is not same, the styles and props content are different